enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/core/consensus/RaftMachine.java

```diff
@@ -214,8 +214,19 @@ public synchronized ExposedRaftState state()
     private void notifyLeaderChanges( Outcome outcome )
     {
         LeaderInfo leaderInfo = new LeaderInfo( outcome.getLeader(), outcome.getTerm() );
+        boolean isStepDown = outcome.isSteppingDown();
+
+        if ( isStepDown )
+        {
+            leaderInfo.stepDown();
+        }
+
         for ( LeaderListener listener : leaderListeners )
         {
+            if ( isStepDown )
+            {
+                listener.onLeaderStepDown( leaderInfo );
+            }
```

Comment:
I'm wondering if it would be better to have a default final method on LeadershipListener called something sensible like `foo` that checks isStepDown, if true calls onLeaderStepDown and then calls onLeaderSwitch.